### ISSUE TITLE
Update Validation for Ozone Season 

### ISSUE THREAD

**Issue by JanellC** [US-EPA-CAMD/PI-1#2536]
### Definition of Ready
- [ ] Does this meet the INVEST criteria?
- [ ] Does this have a clearly defined user?
- [ ] Does this have an estimate?
- [ ] Does this have acceptance criteria?
- [ ] Does this have sufficient context for the specific functionality being developed (design, business rules, etc.)
- [ ] Does this have a dev link been provided in the design?
- [ ] Is this traceable to a feature, epic, or PI?

## User Story
As a CAMPD user I want the ozone season validation to account ozone data only being available between May 1st and Sept 30th so that I can successfully build a query 

## Acceptance Criteria
Given a user is building an Emissions Ozone Query 
When They enter a year that is before 1995
Then the system displays a message indicating that the data is only available between 1995 and the last fully reported ozone season year

Given a user is building an Emissions Ozone Query 
When They enter a year (range) that is between 1995 and the current year on a date before October 1
Then the system displays a message indicating that the data is only available between 1995 and the last fully reported ozone season year

Given a user is building an Emissions Ozone Query 
When They enter a year (range) that is between 1995 and the current year on a date after October 1
Then the system accepts the inputted year
### Development Notes
the system should maintain the error messages it currently has for the appropriate year format (####)

### Does this Story require 508 Testing?
- [ ] Implementing a new authoring tool that will produce code / content for the application
- [ ] Implementing a 3rd party widget that will be part of the application
- [ ] Adding new / updating custom functionality 
- [ ] Adding new / updating a page template
- [ ] Adding new / replacing existing functionality (unless the replacement has already passed 508 testing)
- [ ] Full Site Review prior to a scheduled EPA 508 review
- [ ] Planning a video that will appear on the application
- [ ] Producing a video that will appear on the application
- [ ] Producing Training materials
- [ ] Producing documents

## Context
(User journeys, design mockups, architectural diagrams, business rules, data element definitions)
This update needs to happen for:
- ozone season emissions API
- streaming services ozone season emissions endpoint
- custom data download tool ozone season emissions UI (time period filter)

Currently the time period parameter uses the same validation as the Annual Emissions. This allows a year to be inputted so long as there is reported data for that year (e.g., 1st quarter of 2023 data is reported by end of April 2023, therefore a query with the year 2023 submitted after april 2023 will be successful). However, the ozone season is only from May 1  - Sept 30. While data for the **full** ozone season won't be available until the end of October, data submitted by facilities may begin being available as soon as October 1st (the first day of reporting). Therefore, the time period filter parameter for ozone seasons emissions should not accept the current calendar year until data is available in the system (October 1st).  If a query is inputted and the data is not available, the error message should indicate the user to input a valid year between 1995 and the last ozone season year with data.

See BUG: Endpoints not querying up until 2022 #2501 for background

## Testing
Test Case:
Tester:

### Definition of Done
- [ ] Have all Acceptance Criteria been agreed to and validated by the scrum team?
- [ ] Has the code been unit tested and peer reviewed?
- [ ] Have the functional tests been executed?
- [ ] Have all defects been reviewed and dispositioned, resolved, or deferred?
- [ ] Is the user story ready to be deployed to the test/staging environments?

**Comment by acollad1** [comment]
Need to have Travis review and define definition of done for this ticket

**Comment by ergjustin** [comment]
@tlageman-epa 
First and third Acceptance criteria are working fine now.
I have doubt on second acceptance criteria.
According to my knowledge, the data on ozone season is not accessible until the end of October. For instance, if someone tries to query on October 1st, they should not be able to do so, and an error message display "Enter year(s) between 1995 and and the quarter ending on 09/30/2023"

**Comment by tlageman-epa** [comment]
@ergjustin Following up from our call. here is an updated context paragraph that should cover what we discussed. The one above is worded incorrectly. Thx! 

Currently the time period parameter uses the same validation as the Annual Emissions. This allows a year to be inputted so long as there is reported data for that year (e.g., 1st quarter of 2023 data is reported by end of April 2023, therefore a query with the year 2023 submitted after april 2023 will be successful). However, the ozone season is only from May 1  - Sept 30. While data for the **full** ozone season won't be available until the end of October, data submitted by facilities may begin being available as soon as October 1st (the first day of reporting). Therefore, the time period filter parameter for ozone seasons emissions should not accept the current calendar year until data is available in the system (October 1st).  If a query is inputted and the data is not available, the error message should indicate the user to input a valid year between 1995 and the last ozone season year with data.

**Comment by ergjustin** [comment]
## PR

1. custom data download tool ozone season emissions UI (time period filter): https://github.com/US-EPA-CAMD/easey-campd-ui/pull/613
2. ozone season emissions API: https://github.com/US-EPA-CAMD/easey-emissions-api/pull/423
3. streaming services ozone season emissions endpoint: N/A

**Comment by bchofor** [comment]
Acceptance Criteria
Given a user is building an Emissions Ozone Query
When They enter a year that is before 1995
Then the system displays a message indicating that the data is only available between 1995 and the last fully reported ozone season year.

Navigate to http://localhost:3000/data/custom-data-download
Select Data Type as Emissions & Data Subtype as Ozone Season Emissions and Apply
In filter select TIME PERIOD enter year before 1995 (1994) and Apply filter

**Comment by bchofor** [comment]
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM3h2QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a0a10e7dec86b60fa28d9a1dbba66fd49b88e580/CAMD%20ticket%202.PNG)

**Comment by ergjustin** [comment]
Developer test in dev env
**CAMPD-UI**
[enter a year that is before 1995](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMTl2QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--30e700437f456b2aba8ea0aacf5350d434389481/2536-1994.PNG)
[enter after 1994 to current year October 1](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMkp2QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--ee1ffac3b26a98b80211035026c9ac27bac4aeb4/2536-1995To2023after_oct.PNG)
I am unable to change the system date because I need elevated privilege's. For local development I can able to test this by using custom `new Date('June 25 2023')` function. 

 **emissions-api**
changes does not reflected in dev environment. because only after release v1.3 it will merge in develop branch. Now changes merge with release branch.

**Comment by bchofor** [comment]
**Acceptance Criteria**
Given a user is building an Emissions Ozone Query
When They enter a year (range) that is between 1995 and the current year on a date after October 1
Then the system accepts the inputted year

Navigate to https://campd-dev.app.cloud.gov/data/custom-data-download
Select Data Type as Emissions & Data Subtype as Ozone Season Emissions and Apply
In filter select TIME PERIOD enter a year (range) that is between 1995 and the current year on a date after October 1 and Apply filter

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN2x2QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--df24169cb63ea975b2cd35645bbba03333d8e455/image.png)

**Comment by tlageman-epa** [comment]
Was any further testing done as @ergjustin described above by setting the date manually in staging or is that not possible? That would give me better confidence in accepting this ticket.

**Comment by ergjustin** [comment]
I changed the date and time on my personal computer and tested.
[April-10-2023](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBKzU0QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--62f98f4b22f831b6a01b501addd2768d0c80d12e/April-10-2023.jfif)
[Sept-26-2023](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBKzk0QVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--bd3ad604244d8c08050d2a4f1763fb3cb4b214b8/Sept-26-2023.jfif)

**Comment by tlageman-epa** [comment]
great. thanks for confirming @ergjustin
